src/main/java/edu/cornell/kfs/fp/batch/service/impl/AdvanceDepositServiceImpl.java

```diff
@@ -0,0 +1,1202 @@
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ * <p/>
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ Copyright Indiana University
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU Affero General Public License as
+ published by the Free Software Foundation, either version 3 of the
+ License, or (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU Affero General Public License for more details.
+
+ You should have received a copy of the GNU Affero General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+public class AdvanceDepositServiceImpl implements AdvanceDepositService {
+    private static final Logger LOG = Logger.getLogger(AdvanceDepositServiceImpl.class);
+    private static final String ACH_TRANSACTION_SEQUENCE_NUMBER = "sequenceNumber";
+    private static final int MAX_NOTE_SIZE = 800;
+
+    protected AttachmentService attachmentService;
+    protected BankService bankService;
+    protected BatchInputFileService batchInputFileService;
+    protected BatchInputFileType batchInputFileType;
+    protected BusinessObjectService businessObjectService;
+    protected DateTimeService dateTimeService;
+    protected DocumentService documentService;
+    protected MailService mailService;
+    protected NoteService noteService;
+    protected ParameterService parameterService;
+    protected PersonService personService;
+    protected WorkflowDocumentService workflowDocumentService;
+
+    protected String attachmentsDirectory;
+    private int achTotalPostedTransactions;
+    private int achTotalSkippedTransactions;
+    private int wiredTotalPostedTransactions;
+    private int wiredTotalSkippedTransactions;
+    private KualiDecimal achTotalPostedTransactionAmount;
+    private KualiDecimal achTotalSkippedTransactionAmount;
+    private KualiDecimal wiredTotalPostedTransactionAmount;
+    private KualiDecimal wiredTotalSkippedTransactionAmount;
+
+    /**
+     * @see AdvanceDepositService#createDocuments()
+     */
+    public void createDocuments() {
+        List<AchIncomeTransaction> transactions = (List<AchIncomeTransaction>) businessObjectService.findMatchingOrderBy(AchIncomeTransaction.class, new HashMap(), ACH_TRANSACTION_SEQUENCE_NUMBER, true);
```

Comment:
The HashMap is not using generics, I would recommend adding it unless the method being invoked doesn't work well with generics.